### PR TITLE
[Plugin Actions] Add Action Trigger and Effect API sample plugin

### PR DESCRIPTION
**Pull request by alisonjlai**
simple hello-world plugin sample with `action-trigger` and `action-effect` config type, created with Vite

### CODE REVIEW

File: actions-sample-plugin/vite.config.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import react from '@vitejs/plugin-react';
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  optimizeDeps: {
+    include: ['@sigmacomputing/plugin'],
+  },
+  server: {
+    cors: true,
+  },
```
Comment (peternandersson) [inline actions-sample-plugin/vite.config.ts:11]: rm

File: actions-sample-plugin/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# React + TypeScript + Vite
```
Comment (peternandersson) [inline actions-sample-plugin/README.md:1]: Can we remove this README?

File: actions-sample-plugin/.gitignore [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# Logs
```
Comment (peternandersson) [inline actions-sample-plugin/.gitignore:1]: We don't need to include the gitignore either

Comment (alisonjlai) [reply actions-sample-plugin/.gitignore:1]: I believe this is necessary, otherwise all the node modules will be committed too

**Review by Pearce-Ropion — APPROVED** [review]

**Pull request merged**